crates/revm/src/witness.rs

```diff
@@ -19,6 +19,9 @@ pub struct ExecutionWitnessRecord {
     ///
     /// `keccak(address|slot) => address|slot`
     pub keys: Vec<Bytes>,
+    /// List of block ids for blocks whose hashes were requested during execution
+    /// by the BLOCKHASH opcode.
+    pub block_ids_for_blockhash_opcode: Vec<u64>,
```

Comment:
let's use `BTreeSet` here and change the field name to `block_numbers`, comment should be enough to explain what it represents

Reply:
Yep I was just thinking this -- saves an allcation too